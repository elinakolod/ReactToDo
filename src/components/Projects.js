import React, { Component } from 'react';
import {connect} from 'react-redux';
import {fetchProjectsRequest} from '../redux/actions';

class Projects extends Component {
  componentDidMount = () => {
    this.props.fetchProjectsRequest()
  }

  render() {
    return (
      <div>
        this.props.projects
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    projects: state.projects
  }
}

const mapDispatchToProps = dispatch => ({
  fetchProjectsRequest: () => dispatch(fetchProjectsRequest())
})

export default connect(mapStateToProps, mapDispatchToProps)(Projects);
