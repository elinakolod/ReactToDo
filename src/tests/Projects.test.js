import React from 'react'
import Projects from '../components/Projects'
import {configure, shallow} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import axios from 'axios'
import MockAdapter from 'axios-mock-adapter'

const middlewares = [thunk];
const mockStore = configureStore(middlewares)
const mock = new MockAdapter(axios)

describe('Projects', () => {
  let projects = [{
    id: 1,
    type: 'projects',
    attributes: {
      name: 'test project'
    }
  }]

  const state = {
    entities: {
      projects: {
        0: {id: 1, name: 'Project name', tasks: [{ id: 2, type: 'task' }]}
      },
      tasks: {
        0: {id: 2, name: 'Task name', project_id: 1, comments: [{ id: 3, type: 'comment' }]}
      },
      comments: {
        0: {id: 3, body: 'Comment body', task_id: 1}
      }
    }
  }
  const store = mockStore(state)

  const response = {
    data: projects
  }

  mock.onGet('http://localhost:3000/api/v1/projects').reply(200, response)

  it('matches Projects snapshot', () => {
    const wrapper = renderer.create(<Provider store={store}><Projects /></Provider>)

    expect(wrapper.toJSON()).toMatchSnapshot()
  })
})
