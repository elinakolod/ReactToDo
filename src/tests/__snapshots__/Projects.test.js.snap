// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects matches Projects snapshot 1`] = `
<div>
  <h2>
    Projects
  </h2>
  <hr />
  <br />
  <form
    className=""
    onSubmit={[Function]}
  >
    <div
      className="mb-3 input-group"
    >
      <input
        aria-describedby="basic-addon2"
        aria-label="Project's name"
        className="form-control"
        name="name"
        onChange={[Function]}
        placeholder="Project's name"
        value=""
      />
      <div
        className="input-group-append"
      >
        <button
          className="btn btn-primary"
          disabled={false}
          type="submit"
        >
          Create Project
        </button>
      </div>
    </div>
  </form>
  <br />
  <div>
    <div>
      <span
        id="projectName1"
      >
        <h3
          id="projectHeader1"
        >
          Project name
        </h3>
      </span>
      <a
        href="#"
        onClick={[Function]}
      >
        Edit
      </a>
      <button
        className="btn btn-outline-danger"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Remove
      </button>
      <div
        style={
          Object {
            "paddingTop": "10px",
          }
        }
      >
        <form
          className=""
          onSubmit={[Function]}
        >
          <div
            className="mb-3 input-group"
          >
            <input
              aria-describedby="basic-addon2"
              aria-label="Task's name"
              className="form-control"
              name="new_task_name"
              onChange={[Function]}
              placeholder="Task's name"
              value=""
            />
            <div
              className="input-group-append"
            >
              <button
                className="btn btn-primary"
                disabled={false}
                type="submit"
              >
                Add Task
              </button>
            </div>
          </div>
        </form>
        <div
          className="list-group"
          onKeyDown={[Function]}
        />
      </div>
      <br />
    </div>
  </div>
</div>
`;
